src/components/modals/ManagePlanSchedulingGoalsModal.svelte

```diff
@@ -217,10 +217,13 @@
         ) => {
           const goalId = parseInt(selectedGoalId);
           const isSelected = selectedGoals[goalId];
-          const goalPlanSpec = $allowedSchedulingGoalSpecs[goalId];
+
+          // if we find at least one goal invocation with the selected goal_id, we don't want to insert this goal_id into the plan spec
+          // i.e. this goal was already selected when we entered the modal, so we don't want to kick off an update, which would cause a duplicate invocation to appear
+          const goalAlreadyExistsInPlanSpec = $allowedSchedulingGoalSpecs.find(e => e.goal_id === goalId) !== undefined;
```

Comment:
For V2 it would be nice to have a better UI for adding a second invocation of the same goal from the "Manage" modal... Currently since it's a checkbox, you can only add/remove a single invocation from here & can only add a 2nd via right-click -> "Duplicate Goal"